src/shared/bus-wait-for-jobs.c

```diff
@@ -279,8 +279,15 @@ static int check_wait_response(BusWaitForJobs *d, WaitJobsFlags flags, const cha
                 return -EOPNOTSUPP;
         else if (streq(d->result, "once"))
                 return -ESTALE;
-        else if (STR_IN_SET(d->result, "done", "skipped"))
+        else if (STR_IN_SET(d->result, "done", "skipped")) {
+                if (FLAGS_SET(flags, BUS_WAIT_JOBS_LOG_SUCCESS)) {
+                        if (streq(d->result, "done"))
```

Comment:
You're doing the same string check twice here. Instead, you can split up the `else if` above into two `streq` if statements.